### PR TITLE
Clean up global retention from the cluster state and obsolete transport actions

### PR DESCRIPTION
**Pull request by gmarouli**
In this PR we remove unused code including:

- The transport action for updating, deleting and retrieving the global retention.
- The `DataStreamGlobalRetention` from the cluster state **(this should be bwc safe because we never exposed the APIs to add a data stream lifecycle to the cluster state)**.
- Remove unused privileges monitor and managing global retention.
- Update cluster state update tasks.

The kept `DataStreamGlobalRetentionResolver` considering it could hold and provide the global retention from the settings when we implement it. We just renamed it to `DataStreamGlobalRetentionProvider` for now to better match what it does. 

The factory retention settings should still work after this change.

Changed pages:

- https://elasticsearch_bk_111636.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-privileges.html
- https://elasticsearch_bk_111636.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-api-get-builtin-privileges.html

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java [hunk 1]
```diff
@@ -8,31 +8,25 @@
...
-import java.util.Iterator;
 import java.util.Objects;
 
 /**
  * A cluster state entry that contains global retention settings that are configurable by the user. These settings include:
  * - default retention, applied on any data stream managed by DSL that does not have an explicit retention defined
  * - max retention, applied on every data stream managed by DSL
  */
-public final class DataStreamGlobalRetention extends AbstractNamedDiffable<ClusterState.Custom> implements ClusterState.Custom {
+public final class DataStreamGlobalRetention implements Writeable, ToXContentFragment {
```
Comment (gmarouli) [inline server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java:29]: I could convert this to a record, but I prefer to keep this change as minimal as possible. I can follow up with it at a follow up PR if we keep this class.

Comment (masseyke) [reply server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java:29]: Does this even need to implement ToXContentFragment any more? It's not written to the cluster state, and it's not directly written to a response anywhere, is it?

**Comment by masseyke** [comment]
Looks good to me, with just a couple of comments:
(1) I think we could remove the toXContent stuff from DataStreamGlobalRetention. I don't think it's used, is it? If it is, I don't think it's tested, is it?
(2) In maybe half the places you renamed variables to something like `dataStreamGlobalRetentionProvider` and in half you left them as `dataStreamGlobalRetentionResolver`. It would be good to be consistent (and I favor `dataStreamGlobalRetentionProvider`).

**Review by masseyke — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java (hunk 1 above)
Comment (gmarouli) [reply server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java:29]: Ah good point! I will remove it.

**Comment by gmarouli** [comment]
Update @elastic/es-security I reinstated the privileges and left as noop to avoid breaking serverless tests. If you say it's ok to remove them, I will clean them up.

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Pull request merged**